lib/Mojo/IRC.pm

```diff
@@ -583,7 +583,7 @@ sub _read {
 
     if ($method =~ /^\d+$/) {
       $self->emit("irc_$method" => $message);
-      $method = IRC::Utils::numeric_to_name($method) or return;
+      $method = IRC::Utils::numeric_to_name($method) or next;
```

Comment:
```$method = IRC::Utils::numeric_to_name($method) || $method;``` is just wrong. it would emit the same event twice. Your ```or next``` patch would work just fine.